### PR TITLE
[WYwYHEuA] Update the descriptions of all procedures and functions based on documentation changes

### PR DESCRIPTION
**Pull request by gem-neo4j**

### CODE REVIEW

**Review by Lojjs — CHANGES_REQUESTED** [review]
Thanks for the hard work and a lot of small comment for you! Let me know if we should rather collect all the docs one on a Trello card

File: core/src/main/java/apoc/cypher/CypherFunctions.java [hunk 1]
```diff
@@ -36,14 +36,14 @@ public Object runFirstColumn(String statement, Map<String, Object> params, boole
...
-    public List<Object> runFirstColumnMany(@Name("cypher") String statement, @Name("params") Map<String, Object> params) {
+    @UserFunction("apoc.cypher.runFirstColumnMany")
+    @Description("Runs the given statement with the given parameters and returns the first column collected into a list.")
+    public List<Object> runFirstColumnMany(@Name("statement") String statement, @Name("params") Map<String, Object> params) {
         return (List)runFirstColumn(statement, params, true);
     }
-    @UserFunction
-    @Description("apoc.cypher.runFirstColumnSingle(statement, params) - executes statement with given parameters, returns first element of the first column only, params are available as identifiers")
-    public Object runFirstColumnSingle(@Name("cypher") String statement, @Name("params") Map<String, Object> params) {
+    @UserFunction("apoc.cypher.runFirstColumnSingle")
```
[the remark this answers is not in the document]
Comment (gem-neo4j) [reply core/src/main/java/apoc/cypher/CypherFunctions.java:44]: hmm I can find it, strange

File: core/src/main/java/apoc/nodes/Nodes.java [hunk 2]
```diff
@@ -165,15 +166,15 @@ public Stream<LongResult> delete(@Name("nodes") Object ids, @Name("batchSize") l
         return Stream.of(new LongResult(count));
     }
 
-    @Procedure
-    @Description("apoc.get.rels(rel|id|[ids]) - quickly returns all relationships with these ids")
-    public Stream<RelationshipResult> rels(@Name("relationships") Object ids) {
+    @Procedure("apoc.get.rels")
+    @Description("Returns all relationships with the given ids.")
```
[the remark this answers is not in the document]
Comment (gem-neo4j) [reply core/src/main/java/apoc/nodes/Nodes.java:170]: ooooh this was a very interesting find. So turns out that the exact same proc is in both Extended and Core. In Core it is actually called apoc.nodes.rels and apoc.get.rels in extended, but I assumed wrongly the other way due to the previous description.

I think I need to go through them all and see if this happens elsewhere so we don't accidentally remove the wrong one :O

File: core/src/main/java/apoc/coll/Coll.java [hunk 3]
```diff
@@ -425,8 +426,8 @@ public List<Object> insert(@Name("coll") List<Object> coll, @Name("index") long
         return list;
     }
 
-    @UserFunction
-    @Description("apoc.coll.insertAll(coll, index, values) | insert values at index")
+    @UserFunction("apoc.coll.insertAll")
+    @Description("Inserts all of the values into the list, starting at the specified index.")
     public List<Object> insertAll(@Name("coll") List<Object> coll, @Name("index") long index, @Name("values") List<Object> values) {
```
[the remark this answers is not in the document]
Comment (Lojjs) [reply core/src/main/java/apoc/coll/Coll.java:431]: Should be fixed in docs as well

File: core/src/main/java/apoc/coll/Coll.java [hunk 4]
```diff
@@ -476,9 +477,9 @@ public boolean containsSorted(@Name("coll") List<Object> coll, @Name("value") Ob
 //        boolean result = (list.size() < batchSize) ? contains.test(list) : partitionList(list, batchSize).parallel().anyMatch(contains);
     }
 
-    @UserFunction
-    @Description("apoc.coll.containsAllSorted(coll, value) optimized contains-all on a sorted list operation (Collections.binarySearch) (returns single row or not)")
-    public boolean containsAllSorted(@Name("coll") List<Object> coll, @Name("values") List<Object> values) {
+    @UserFunction("apoc.coll.containsAllSorted")
+    @Description("Returns whether or not the given value exists in an already sorted collection (using a binary search).")
```
[the remark this answers is not in the document]
Comment (Lojjs) [reply core/src/main/java/apoc/coll/Coll.java]: Should be fixed in docs as well

File: core/src/main/java/apoc/number/Numbers.java [hunk 5]
```diff
@@ -19,7 +19,7 @@
 public class Numbers {
 
   @UserFunction("apoc.number.format")
-  @Description("Formats the given long or double using the default system pattern and language to produce a string.")
+  @Description("Formats the given long or double using the given pattern and language to produce a string.")
```
Comment (Lojjs) [inline core/src/main/java/apoc/number/Numbers.java:22]: The 3 changes in this file should be made to docs as well

File: core/src/main/java/apoc/lock/Lock.java [hunk 6]
```diff
@@ -43,16 +43,16 @@ public void readLockOnNodes(@Name("nodes") List<Node> nodes) {
         }
     }
 
-    @Procedure(name = "apoc.lock.read.rels", mode = Mode.WRITE)
-    @Description("Acquires a read lock on the given relationships.")
+    @Procedure(name = "apoc.lock.nodes", mode = Mode.WRITE)
```
Comment (Lojjs) [inline core/src/main/java/apoc/lock/Lock.java:46]: I think this one should be named apoc.lock.rels

**Review by Lojjs — APPROVED** [review]

**Pull request merged**